wagtail/search/backends/elasticsearch6.py

```diff
@@ -12,6 +15,11 @@ class Elasticsearch6Mapping(Elasticsearch5Mapping):
     all_field_name = "_all_text"
     edgengrams_field_name = "_edgengrams"
 
+    def get_boost_field_name(self, boost):
+        # replace . with _ to avoid issues with . in field names
+        boost = str(float(boost)).replace(".", "_")
+        return f"{self.all_field_name}_boost$$${boost}"
```

Comment:
suggestion: add a note as to the use of `$$$`

Reply:
I'm not really keen on this pattern of encoding the boost value in a specially-crafted field name so that we can decode it later on when applying the boost... it feels like a "code smell" that we're having to repurpose a name to store information that we should have access to already.

I think the best way around that is possibly to do some refactoring so that instead of passing field names around in `remapped_fields` and `get_field_column_name`, we introduce a "Column" object that has name and boost attributes and pass those around instead. I'll do some further investigation and perhaps open a separate cleanup PR...

Reply:
@KIRA009 That would be great, thanks!